compiler/src/iree/compiler/Dialect/Flow/Conversion/TensorToFlow/Patterns.cpp

```diff
@@ -183,6 +183,56 @@ struct ConvertTensorFromElementsPattern
   }
 };
 
+/// Populates given sizes array with the dynamic dims.
+static void sizesForTensor(OpBuilder &builder, SmallVectorImpl<Value> &sizes,
+                           Location loc, ShapedType stp, Value tensor) {
+  for (const auto &d : enumerate(stp.getShape())) {
+    if (d.value() == ShapedType::kDynamic) {
+      Value dim = builder.create<tensor::DimOp>(loc, tensor, d.index());
+      sizes.push_back(dim);
+    }
+  }
+}
+
+/// Convert tensor.reshape ops into flow.tensor.reshape ops where possible.
+struct ConvertTensorDialectReshapeOpPattern
+    : public OpRewritePattern<tensor::ReshapeOp> {
+  using OpRewritePattern<tensor::ReshapeOp>::OpRewritePattern;
+  LogicalResult matchAndRewrite(tensor::ReshapeOp op,
+                                PatternRewriter &rewriter) const override {
+      if (op->getParentOfType<Flow::DispatchWorkgroupsOp>()) {
+        return failure();
+      }
+      auto loc = op.getLoc();
+      Value input = op.getSource();
+      ShapedType inputType = dyn_cast<ShapedType>(input.getType());
+      ShapedType shapeOperandType = dyn_cast<ShapedType>(op.getShape().getType());
+      ShapedType resultType = dyn_cast<ShapedType>(op.getResult().getType());
```

Comment:
style nit: LLVM style guide states that we use `auto` when the type is obvious from the context (i.e., dyn_cast<...>). 

https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable